project-loader/src/main/kotlin/Args.kt

```diff
@@ -12,8 +12,19 @@ private fun <T> splitAndCreate(str: String, creator: (String, String) -> T): T {
     return creator(split[0], split[1])
 }
 
+private fun <T> splitAndCreatePlugin(str: String, creator: (String, String, Boolean) -> T): T {
+    val split = str.split("::")
+    if (split.size < 2) {
+        throw RuntimeException("string is not of the right format. Expected <key>::<value>[::<pre installed>]")
+    }
+    if (split.size == 3){
+        return creator(split[0], split[1], split[2].toBoolean())
```

Comment:
we could use the "elvis operator" here to check the parse result and otherwise just make preinstalled false? Like this: 
```suggestion
        return creator(split[0], split[1], split[2].toBoolean() ?: false)
```

That was the last argument doesn't have to be nullable.